monitoring/thread_status_updater.cc

```diff
@@ -182,7 +203,7 @@ Status ThreadStatusUpdater::GetThreadList(
     uint64_t op_props[ThreadStatus::kNumOperationProperties] = {0};
 
     auto iter = cf_info_map_.find(cf_key);
-    if (iter != cf_info_map_.end()) {
```

Comment:
Honestly I think `SetColumnFamilyInfoKey` mainly serves `cf_info_map_`, the same purpose as relevant book-keeping data structures/functions e.g, `db_key_map_`, `EraseDatabaseInfo` and `EraseColumnFamilyInfo` as well as determining whether we should enable tracking based on whether we have `cf_key`. 

`cf_info_map_` ultimately is used to skip some thread status info population here to achieve the result of making these thread seem "useless/lacking info" if `Env::GetThreadList()` is called after calling db close or dropping cf on Env object. 

I highly doubt whether this is necessary - but I don't intend to make the optimization for that in this PR. So currently I decided to follow its "overkill-game" and add a db_name field for db-scope thread like the one that does db open and then list this thread out too (but later filter it in Env-level get thread list). I think a good timing to improve this is when we want to expose user thread just as bg thread in a similar way as Env::GetThreadList() for non-testing purpose.